MonoGame.Framework/Graphics/TextureCollection.cs

```diff
@@ -13,14 +13,14 @@ public sealed partial class TextureCollection
     {
         private readonly GraphicsDevice _graphicsDevice;
         private readonly Texture[] _textures;
-        private readonly bool _applyToVertexStage;
+        private ShaderStage _stage;
         private int _dirty;
 
-        internal TextureCollection(GraphicsDevice graphicsDevice, int maxTextures, bool applyToVertexStage)
+        internal TextureCollection(GraphicsDevice graphicsDevice, int maxTextures, ShaderStage stage)

```

Comment:
This seems like a better approach considering we may eventually have more shader stages.